apps/analysis/tests/test_apis.py

```diff
@@ -89,6 +89,62 @@ def test_create_pillar_from_analysis(self):
         self.assert_201(response)
         self.assertEqual(AnalysisPillar.objects.count(), pillar_count + 1)
 
+    def test_create_pillar_along_with_statement(self):
+        pillar_count = AnalysisPillar.objects.count()
+        statement_count = AnalyticalStatement.objects.count()
+        user = self.create_user()
+        project = self.create_project()
+        project.add_member(user)
+        entry1 = self.create(Entry)
+        entry2 = self.create(Entry)
+        analysis = self.create(Analysis, title='Test Analysis')
+        url = f'/api/v1/projects/{project.id}/analysis/{analysis.id}/pillars/'
+        data = {
+            'main_statement': 'Some main statement',
+            'information_gap': 'Some information gap',
+            'assignee': user.id,
+            'title': 'Some title',
+            'analytical_statement': [
+                {
+                    "statement": "coffee",
+                    "order": 1,
+                    "analytical_entries": [
+                        {
+                            "order": 1,
+                            "entry": entry1.id,
+                        },
+                        {
+                            "order": 2,
+                            "entry": entry2.id
+                        }
+                    ],
+                }
+            ]
+        }
+        self.authenticate(user)
+        response = self.client.post(url, data)
+        self.assert_201(response)
+        self.assertEqual(AnalysisPillar.objects.count(), pillar_count + 1)
+        self.assertEqual(AnalyticalStatement.objects.filter(
+                         analysis_pillar__analysis=analysis).count(), statement_count + 1)
+
+        # try to edit
+        response_id = response.data['id']
+        data = {
+            'main_statement': 'HELLO FROM MARS',
+            'analytical_statement': [
+                {
+                    'statement': "tea",
+                    'order': 2
+                }
+            ]
+        }
+        self.authenticate(user)
+        url = f'/api/v1/projects/{project.id}/analysis/{analysis.id}/pillars/{response_id}/'
+        response = self.client.patch(url, data)
+        self.assert_200(response)
+        self.assertEqual(response.data['main_statement'], data['main_statement'])
```

Comment:
Add test:

Check if old analytical statements is/should be retained.
Check new analytical statements count.

Reply:
Done, thanks !!